src/pip/_internal/commands/show.py

```diff
@@ -119,7 +119,7 @@ def _get_requiring_packages(current_dist: BaseDistribution) -> Iterator[str]:
         yield _PackageInfo(
             name=dist.raw_name,
             version=str(dist.version),
-            location=dist.location or "",
+            location=dist.editable_project_location or dist.location or "",
```

Comment:
I think this should be a new field, since location does have a valid meaning for editables too and to avoid mixing different semantics in the same field.

Reply:
this is the location shown in `pip list` and that's the actual location from the user's point of view.
what do you suggest as the other field? i think showing an additional `editable location` will just make it look confusing

Reply:
> this is the location shown in pip list and that's the actual location from the user's point of view.

In pip list, the column title is `Editable project location` and is populated for editables only.
`pip list -v` shows an additional `Location` column with `dist.location` which is where the project metadata is (which is different from the project location for PEP 660 editables).
It is similar in `pip inspect` and `pip list --format=json`.

So in pip show, it makes sense to show a new `Editable project location` field.

Reply:
changed